### PR TITLE
feat(HACBS-1386): parse hacbs test result file

### PR DESCRIPTION
**Pull request by hongweiliu17**
Use shell function parse_hacbs_test_output to parse hacbs test result file

Signed-off-by: Hongwei Liu hongliu@redhat.com

### CODE REVIEW

**Comment by jsztuka** [comment]
I suppose this change could be merged only after https://github.com/redhat-appstudio/hacbs-test/pull/34 gets merged first.

**Comment by hongweiliu17** [comment]
> I suppose this change could be merged only after [redhat-appstudio/hacbs-test#34](https://github.com/redhat-appstudio/hacbs-test/pull/34) gets merged first.

Yes.

File: task/clair-scan/0.1/clair-scan.yaml [hunk 1]
```diff
@@ -43,22 +43,12 @@ spec:
...
         HACBS_ERROR_OUTPUT=$(jq -rc --arg date $(date +%s) --null-input \
           '{result: "ERROR", timestamp: $date}')
-        HACBS_TEST_OUTPUT=$(jq -rce --arg date $(date +%s) \
-          '.[] | { result: (if (.failures | length > 0) then "FAILURE" else "SUCCESS" end),
-                   timestamp: $date,
-                   namespace,
-                   successes,
-                   failures: (.failures | length)
-                 }' /tekton/home/clair-vulnerabilities.json || true)
+        HACBS_TEST_OUTPUT=
```
Comment (Michkov) [inline task/clair-scan/0.1/clair-scan.yaml:49]: Probably offtopic but should not we stop using HACBS keyword? So uste TEST_OUTPUT?

Comment (dirgim) [reply task/clair-scan/0.1/clair-scan.yaml:49]: That is a good point, but we will need to do that in concert with EC, so we're expecting to have collaborative story/ies to make the naming changes.

File: task/sast-snyk-check/0.1/sast-snyk-check.yaml [hunk 2]
```diff
@@ -48,21 +48,9 @@ spec:
...
           cat sast_snyk_check_out.json
-          HACBS_TEST_OUTPUT=$(jq -rce --arg date $(date +%s) \
-            '{ result: (if (.runs[].results | length > 0) then "FAILURE" else "SUCCESS" end),
-                   timestamp: $date,
-                   namespace: "default",
-                   successes: 0,
-                   note: "",
-                   failures: (.runs[].results | length)
-                 }' sast_snyk_check_out.json || true)
+          HACBS_TEST_OUTPUT=
```
Comment (dirgim) [inline task/sast-snyk-check/0.1/sast-snyk-check.yaml:51]: I think this file is missing the `. /utils.sh` call.

Comment (hongweiliu17) [reply task/sast-snyk-check/0.1/sast-snyk-check.yaml:52]: Yes, thanks!

Comment (hongweiliu17) [reply task/sast-snyk-check/0.1/sast-snyk-check.yaml:52]: Updated

**Review by jsztuka — APPROVED** [review]
lgtm

**Review by dirgim — APPROVED** [review]
LGTM

**Review by srivickynesh — APPROVED** [review]
LGTM

File: task/clair-scan/0.1/clair-scan.yaml (hunk 1 above)
Comment (hongweiliu17) [reply task/clair-scan/0.1/clair-scan.yaml:49]: Thank you. Let's update them to the name without `HACBS` once we make this decision in the future.

**Pull request merged**